### PR TITLE
Bump `chrono` from 0.4.34 to 0.4.38

### PR DESCRIPTION
**Pull request by upbqdn**
## Motivation & Solution

- Bump `chrono` from 0.4.34 to 0.4.38.
- Tests:
  - Remove a deprecated fn.
  - Minor cleanups.

#### Check before marking the PR as ready for review:
  - [x] Will the PR name make sense to users?
  - [x] Does the PR have a priority label?
  - [x] Have you added or updated tests?
  - [x] Is the documentation up to date?

### Reviewer Checklist

Check before approving the PR:
  - [ ] Does the PR scope match the ticket?
  - [ ] Are there enough tests to make sure it works? Do the tests cover the PR motivation?
  - [ ] Are all the PR blockers dealt with?
        PR blockers can be dealt with in new tickets or PRs.

_And check the PR Author checklist is complete._

### CODE REVIEW

**Comment by arya2** [comment]
@gustavovalverde It looks like the firebase login is failing: https://github.com/ZcashFoundation/zebra/actions/runs/8691548411/job/23834074763?pr=8400#step:8:73

File: zebra-chain/src/block/tests/generate.rs [hunk 1]
```diff
@@ -182,8 +170,8 @@ fn single_transaction_block_many_inputs(oversized: bool) -> Block {
     // A block header
     let (block_header, block_header_bytes) = block_header();
 
-    // A LockTime
-    let (lock_time, lock_time_bytes) = lock_time();
+    let lock_time = LockTime::Time(DateTime::from_timestamp(61, 0).unwrap());
```
Comment (arya2) [inline zebra-chain/src/block/tests/generate.rs:173]: Optional/Nitpick: Do we want to move this to a constant?

**Review by arya2 — DISMISSED** [review]

File: zebra-chain/src/block/tests/generate.rs (hunk 1 above)
Comment (upbqdn) [reply zebra-chain/src/block/tests/generate.rs:173]: I don't think so. It's an arbitrarily picked dummy value used only in test code, without much significance.

**Comment by gustavovalverde** [comment]
> @gustavovalverde It looks like the firebase login is failing: https://github.com/ZcashFoundation/zebra/actions/runs/8691548411/job/23834074763?pr=8400#step:8:73

Thank you @arya2 ! This was a typo while I was fixing that production project. It should be good now.

File: zebra-chain/src/block/tests/generate.rs [hunk 2]
```diff
@@ -256,8 +244,8 @@ fn single_transaction_block_many_outputs(oversized: bool) -> Block {
...
```
Comment (arya2) [inline zebra-chain/src/block/tests/generate.rs:247]: Optional/Nitpick: Do we still want to use the updated method if the deprecated one has been restored?

**Review by arya2 — APPROVED** [review]
Looks good!

File: zebra-chain/src/block/tests/generate.rs (hunk 2 above)
Comment (upbqdn) [reply zebra-chain/src/block/tests/generate.rs:247]: The removed `lock_time` fn was ours. It used the `from_timestamp_opt` fn, which is deprecated even in the newest version of `chrono`, and the recommended one is `from_timestamp`, which we're using. The deprecation here means that there's now a warning when using `from_timestampt_opt`.

The other deprecated fns in the previous version of `chrono` were actually removed, and Zebra wouldn't compile. The newer version has them back without deprecation warnings.

**Comment by oxarbitrage** [comment]
@Mergifyio update

**Pull request merged**